test/SerilogTracing.Tests/LoggerTracingExtensionsTests.cs

```diff
@@ -85,11 +84,10 @@ public void ActivityIsGeneratedWhen(bool levelEnabled, bool tracingEnabled, bool
             Assert.Equal(parent.SpanId,
                 (span.Properties[Core.Constants.ParentSpanIdPropertyName] as ScalarValue)?.Value);
         }
-        else
-        {
-            Assert.Same(LoggerActivity.None, activity);
-            Assert.Empty(sink.Events);
-        }
+        
+        // NOTE: We can't assert there's no activity when `activityExpected` is `false`,
+        // because we use a shared `ActivitySource`, any other listener in a concurrent test
+        // run may still cause activities to be created despite sampling
```

Comment:
It's possible to mark the affected test classes with `[Category("Shared")]` to enforce that they run serially.

Reply:
I'll give that a go :+1: It's a problem for any tests that use the full pipeline.